contracts/StablecoinConverter.sol

```diff
@@ -175,18 +186,35 @@ contract StablecoinConverter is EpochTokenLocker {
                 volumes[i]
             );
         }
+        int fee = tokenConservation[0];
+        require(fee < auctionData[batchIndex].feeCollected, "Fee is not higher than before");
+        auctionData[batchIndex].feeCollected = fee;
+        auctionData[batchIndex].bestSolutionSubmitter = msg.sender;
         checkTokenConservation(tokenConservation);
         documentTrades(batchIndex, owners, orderIds, volumes, tokenIdsForPrice);
     }
 
     function checkTokenConservation(
         int[] memory tokenConservation
     ) internal pure {
-        for (uint i = 0; i < tokenConservation.length; i++) {
+        for (uint i = 1; i < tokenConservation.length; i++) {
             require(tokenConservation[i] == 0, "Token conservation does not hold");
         }
     }
 
+    function updateCurrentPrices(
+        uint128[] memory prices,  //list of prices for touched token only, frist price is fee Token price
```

Comment:
nit: could we align this comment with the one in the next line? Also, "touched tokens" - plural.